cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -162,19 +167,36 @@ void DrawableSubmap::SetAlpha(const double current_tracking_z) {
       target_alpha == 0.f || target_alpha == 1.f) {
     current_alpha_ = target_alpha;
   }
-  ogre_slice_.SetAlpha(current_alpha_);
+  for (auto& slice : ogre_slices_) {
+    slice->SetAlpha(current_alpha_);
+  }
   display_context_->queueRender();
 }
 
+void DrawableSubmap::SetSliceVisibility(size_t slice_index, bool visible) {
+  if (slice_index < slices_visibility_.size()) {
+    slices_visibility_[slice_index] = visible;
+  }
+  ToggleVisibility();
+}
+
 void DrawableSubmap::UpdateSceneNode() {
   ::cartographer::common::MutexLocker locker(&mutex_);
-  // TODO(gaschler): Add UI feature to show all textures.
-  ogre_slice_.Update(submap_textures_->textures[0]);
+  for (size_t slice_index = 0; slice_index < ogre_slices_.size() &&
```

Comment:
do we ever expect to get a submap_textures_.size() != ogre_slices_? I think this would be invalid data, dropped earlier. So CHECK this here?

Reply:
2D will continue to send one slice.